tuplex/core/src/DataSet.cc

```diff
@@ -760,6 +756,115 @@ namespace tuplex {
         printTable(os, headers, rows);
     }
 
+    void printHTMLRow(std::ostream &os, size_t ind, const Row& r) {
+        os << "    <tr>\n";
+        os << fmt::format("      <th>{}</th>\n", ind);
+        for (auto& s : r.getAsStrings()) {
+            os << fmt::format("      <td>{}</td>\n", s);
+        }
+        os << "    </tr>\n";
+    }
+
+    void DataSet::showHTMLPreview(size_t topLimit, size_t bottomLimit, std::ostream &os) {
+        std::string HTML_TEMPLATE =
```

Comment:
I feel like this should be split into a function that does produce the HTML preview table and one that gets the result set, i.e.

```
showAsHTML(topLimit, bottomLimit, os) {
auto rs = take(topLimit, bottomLimit);
os<<resultSetAsHTMLTable(rs);
}
```

Reply:
Also, why not move this logic to python? Is there a reason why it's in C++?